### PR TITLE
[infer] fix test bug

### PR DESCRIPTION
**Pull request by Xu-Kai**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?
- Fix llama, bloom, chatglm2 test bug.
- Fix llama2 attention test bug.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [x] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: tests/test_infer/test_llama_infer.py [hunk 1]
```diff
@@ -53,22 +54,24 @@ def init_to_get_rotary(self, base=10000):
...
-        shard_config = ShardConfig(
-            enable_tensor_parallelism=True if test_config["tp_size"] > 1 else False, inference_only=True
-        )
-        infer_engine = TPInferEngine(orig_model, shard_config, BATCH_SIZE, MAX_INPUT_LEN, MAX_OUTPUT_LEN)
-
-        generate_kwargs = dict(do_sample=False)
-        outputs = infer_engine.generate(data, **generate_kwargs)
-
-        assert outputs is not None
+    llama_config = LlamaConfig(num_hidden_layers=2, bos_token_id=0, eos_token_id=1, vocab_size=1200, hidden_size=1024)
```
Comment (tiandiao123) [inline tests/test_infer/test_llama_infer.py:57]: why changed like this? no use model zoo?

**Review by tiandiao123 — APPROVED** [review]

File: tests/test_infer/test_llama_infer.py (hunk 1 above)
Comment (Xu-Kai) [reply tests/test_infer/test_llama_infer.py:57]: model zoo can't pass the test.

**Review by yuehuayingxueluo — APPROVED** [review]

File: tests/test_infer/test_chatglm2_infer.py [hunk 2]
```diff
@@ -31,28 +29,31 @@
...
 )
 def run_chatglm2_test(test_config):
-    tokenizer = AutoTokenizer.from_pretrained("THUDM/chatglm2-6b", trust_remote_code=True)
-    # pad_token_id = 0
-    model_fn = lambda: ChatGLMForConditionalGeneration(infer_config, empty_init=False)
-    orig_model = model_fn()
-    orig_model = orig_model.half()
-    text = ["how is the weather today?"]
-    input_ids = tokenizer.batch_encode_plus(text, return_tensors="pt", padding=True)
+    chagglm_config = ChatGLMConfig(
```
Comment (CjhHa1) [inline tests/test_infer/test_chatglm2_infer.py:32]: typo

Comment (Xu-Kai) [reply tests/test_infer/test_chatglm2_infer.py]: Done.

**Pull request merged**